api.py

```diff
@@ -33,3 +33,15 @@ def wrapper(*args, **kwargs):
         except discover.UnknownIdError:
             flask.abort(404)
     return wrapper
+
+
+@app.route('/preparations/')
+@app.route('/preparations/<int:dataset_id>')
+@missing_key_returns_404
+def preparations(dataset_id: int=None):
```

Comment:
Is it python standard to have route to all preparation and with id together? Maybe is it better to split it?

Reply:
I've tried, but Flask works in such a manner, that it groups up single route into a single function (view) and requires it to be a single element. Since I've found no way of doing it, I've migrated any other logic back, while here we have only this distinction.